### DISCUSSION TITLE
V8 엔진 최적화를 통한 실시간 라이브 강의 플랫폼의 비디오 및 채팅 지연 이슈 해결

### DISCUSSION THREAD

**Discussion by ehvzmf** [100-hours-a-week/discussions#15] [Deepdive]
##### 시나리오:
사용자가 실시간으로 라이브 강의를 시청하면서 원활한 비디오 스트리밍과 즉각적인 질문/응답 기능을 이용할 때, JavaScript 성능 최적화는 필수적입니다. V8 엔진은 이러한 시나리오에서 중요한 역할을 합니다. 예를 들어, 강의 도중 다수의 사용자가 실시간 채팅을 통해 질문을 던지면, V8 엔진은 이 데이터를 신속하게 처리하고 메모리 관리를 효율적으로 수행해야 합니다.

1. **비디오 스트리밍 지연 및 끊김**: 강의를 시청하던 사용자가 갑자기 비디오가 끊기거나 몇 초씩 지연되는 현상을 겪을 수 있습니다. 이는 WebRTC를 통한 실시간 비디오 데이터 처리 중 V8 엔진이 JavaScript 코드를 비효율적으로 실행하거나, 과도한 Garbage Collection (GC)으로 인해 브라우저가 일시적으로 멈추면서 발생하는 문제입니다.

2. **실시간 채팅의 응답 지연**: 사용자가 질문을 입력하고 강사 또는 다른 사용자로부터 답변을 받기까지 시간이 오래 걸리는 경우가 있습니다. 이러한 지연은 웹소켓을 통한 메시지 전달 과정에서 V8 엔진이 JavaScript 코드를 최적화하지 못해 발생하며, 특히 많은 사용자들이 동시에 메시지를 전송할 때 문제가 심화됩니다.

3. **브라우저 반응성 저하**: 강의 중 사용자가 웹 페이지 내 다른 기능(예: 노트 작성, 퀴즈 참여 등)을 사용하려 할 때, 브라우저가 느려지고 클릭이나 입력이 즉각 반응하지 않는 문제가 생길 수 있습니다. 이는 V8 엔진이 실행하는 JavaScript가 많은 연산을 요구하거나, 메모리 관리가 제대로 이루어지지 않아 브라우저 전체의 성능이 떨어질 때 발생합니다.

이러한 문제점들을 해결하기 위해, V8 엔진이 JavaScript 코드를 어떻게 최적화하여 실행하는지에 대해 깊이 탐구하고자 합니다. 궁극적으로, 실시간 라이브 강의 플랫폼에서 발생할 수 있는 성능 저하 문제를 개선하여 사용자들이 끊김 없는 비디오 스트리밍과 즉각적인 채팅 응답을 경험할 수 있도록 하는 것이 목표입니다.

**Comment by startupcodekr** [comment]
이유를 제목에 녹여주세요!

**Reply by startupcodekr** [reply]
또한 어떤 포인트에서 v8엔진하고 사용 기술(웹소켓, webrtc)와 상관있는지 더 자세히 기술 필요합니다.
'Garbage Collection (GC)와 메모리 관리', 'JavaScript 코드 실행 효율성' 등

**Reply by ehvzmf** [reply]
반영해보았습니다.
